### DISCUSSION TITLE
No way to change smart-quote logic to other languages?

### DISCUSSION THREAD

**Discussion by kolstadmagnus** [gatsbyjs/gatsby#33704] [Help]
I submitted [this question](https://stackoverflow.com/questions/69613921/how-do-i-change-smart-quote-language-in-gatsby-js) on Stack Overflow about a week ago. I asked if there was a way to change Gatsby’s smart-quote substitution logic to another language. I didn’t receive any responses of substance, so I assume this isn’t possible in Gatsby?

That seems strange to me, given the [great typographic diversity from language to language when it comes to quotation marks](https://en.wikipedia.org/wiki/Quotation_mark#Summary_table). It’s also puzzling seeing as Gatsby already has [this plugin](https://www.gatsbyjs.com/plugins/gatsby-remark-smartypants/)—it should be easy to allow options to swap the American double quotes `“` `”` with, e.g., French guillemets `«` `»`, no?

**Comment by SimpleCreations** [comment]
I am facing the same issue. Have you found a solution?

**Comment by LekoArts** [comment]
It's hardcoded in `retext-smartypants`:

https://github.com/retextjs/retext-smartypants/blob/aae96eeafcbc1e98413c6ef939108cd728af699e/index.js#L53-L54

You could copy/paste that plugin into your own local remark plugin and change the logic there.

**Comment by awesomephant** [comment]
Looks like `retext-smartypants` supports configurable quotation marks in [5.2](https://github.com/retextjs/retext-smartypants/releases/tag/5.2.0). However, `gatsby-remark-smartypants` is pinned to `^4.0.0` - possibly because later releases use ESM? @LekoArts would you accept a pull request on this or are dependency updates handled differently?

**Reply by LekoArts** [reply]
It's ESM-only and right now we don't have our favourite way of handling that figured out yet. For now I'd recommend [creating a local plugin](https://www.gatsbyjs.com/docs/creating-a-local-plugin/), then using [`gatsby-config.mjs`](https://www.gatsbyjs.com/docs/reference/release-notes/v5.3/#es-modules-esm-in-gatsby-files) and authoring your remark plugin in ESM.

**Reply by awesomephant** [reply]
I just tried this, but it looks like Gatsby is still trying to `require()` my local plugin written in ESM: `Error [ERR_REQUIRE_ESM]: require() of ES Module ...\my-site\plugins\my-plugin\index.js from ...\my-site\node_modules\gatsby\dist\bootstrap\load-plugins\validate.js not
supported`. I'm using `gatsby-config.mjs` and I'm loading the plugin as described [here](https://www.gatsbyjs.com/docs/creating-a-local-plugin/#project-structure-for-a-local-plugin), am I missing something?

**Reply by LekoArts** [reply]
Try `gatsby@next` version, I think we didn't ship a fix for that to latest yet

**Reply by awesomephant** [reply]
Did that, now I'm getting: `Only URLs with a scheme in: file, data are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'd:' (gatsbyRemarkPlugins)`. Any ideas?

**Reply by LekoArts** [reply]
Ah... https://github.com/gatsbyjs/gatsby/pull/37520 should fix it

**Reply by awesomephant** [reply]
This works 🎉. Unfortunately guillemets are only half-viable because `retext-smartypants` doesn't distinguish between closing single quotation marks and apostrophes, so `Max's house` turns into `Max‹s house`, but that's an issue with the underlying library and not the Gatsby integration.
